### PR TITLE
catalyst-browser: init at 3.9.4

### PR DESCRIPTION
**Pull request by jdev082**
Adds [Catalyst](https://getcatalyst.eu.org) to the package repository.

Source code is [here](https://github.com/CatalystDevOrg/Catalyst)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jdev082** [comment]
I'm trying to format the file with the format tool and it's not doing anything.

*I am in the directory of the package.nix file*

```
$ nixpkgs-fmt ./package.nix

0 / 1 have been reformatted
```

**Comment by FliegendeWurst** [comment]
You need to go to the Nixpkgs root directory, run `nix-shell`, then:
nixfmt 'pkgs/by-name/ca/catalyst/package.nix'

File: pkgs/by-name/ca/catalyst/package.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  postFixup = ''
+    makeWrapper ${electron}/bin/electron $out/bin/${pname} \
+      --add-flags $out/share/${pname}/resources/app.asar \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ stdenv.cc.cc ]}"
+  '';
+
+  meta = with lib; {
+    description = "A minimal, functional, and customizable user-focused FOSS web browser ";
+    homepage = "https://getcatalyst.eu.org";
+    license = licenses.mit;
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/catalyst/package.nix:80]:
```suggestion
  meta = {
    description = "Minimal, functional, and customizable user-focused FOSS web browser based on QtWebEngine";
    homepage = "https://getcatalyst.eu.org";
    license = lib.licenses.mit;
```

File: pkgs/by-name/ca/catalyst/package.nix [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace-fail 'Exec=AppRun' 'Exec=${pname}'
+
+    asar pack resources/ $out/share/${pname}/resources/app.asar
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    makeWrapper ${electron}/bin/electron $out/bin/${pname} \
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/catalyst/package.nix:72]:
```suggestion
    makeWrapper ${electron}/bin/electron $out/bin/${meta.mainProgram} \
```

File: pkgs/by-name/ca/catalyst/package.nix [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    rm -rf resources/contributing.md
+    rm -rf resources/pnpm-lock.yaml
+    rm -rf resources/README.md
+    rm -rf resources/CODE_OF_CONDUCT.md
+    rm -rf *.nix
+    substituteInPlace resources/src/index.html \
+      --replace-fail 'catalyst-default-distrib' 'catalyst-default-nixpkgs'
+
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace-fail 'Exec=AppRun' 'Exec=${pname}'
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/catalyst/package.nix:64]:
```suggestion
      --replace-fail 'Exec=AppRun' 'Exec=${meta.mainProgram}'
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ca/catalyst/package.nix (hunk 1 above)
Comment (jdev082) [reply pkgs/by-name/ca/catalyst/package.nix:80]: The Catalyst 5 Alpha uses QtWebEngine, this is Catalyst 3. I will make the other requested changes.

**Comment by jdev082** [comment]
It seems like there's some font problems. I searched all the fonts that Catalyst checks for but couldn't find any of them in nixpkgs. 

```

**Comment by jdev082** [comment]
Requested changes applied and package updated to latest version.

File: pkgs/by-name/ca/catalyst/package.nix (hunk 1 above)
Comment (FliegendeWurst) [reply pkgs/by-name/ca/catalyst/package.nix]: Ah I see. In that case a more precise homepage value may also be good.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by jdev082** [comment]
This pull request is ready unless there's any more requested changes. Aarch64 Darwin binaries are officially available and could be added.

File: pkgs/by-name/ca/catalyst/package.nix [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ stdenv.cc.cc ]}"
+  '';
+
+  meta = {
+    description = "Minimal, functional, and customizable user-focused FOSS web browser based on Chromium";
+    homepage = "https://getcatalyst.eu.org";
+    license = lib.licenses.mit;
+    mainProgram = "catalyst";
+    maintainers = with lib.maintainers; [ jdev082 ];
+    platforms = [ "x86_64-linux" ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/catalyst/package.nix:84]:
```suggestion
    platforms = [ "x86_64-linux" ];
    sourceProvenance = [ lib.sourceTypes.binaryNativeCode ];
```

File: pkgs/by-name/ca/catalyst/package.nix [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    rm -rf resources/.pnpm-debug.log
+    rm -rf resources/contributing.md
+    rm -rf resources/pnpm-lock.yaml
+    rm -rf resources/README.md
+    rm -rf resources/CODE_OF_CONDUCT.md
+    rm -rf *.nix
+    substituteInPlace resources/src/index.html \
+      --replace-fail 'catalyst-default-distrib' 'catalyst-default-nixpkgs'
+
+    substituteInPlace $out/share/applications/${pname}.desktop \
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/catalyst/package.nix:64]: Please avoid using pname here and elsewhere in the package.

#277994

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ca/catalyst/package.nix [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  pname = "catalyst";
+  version = "3.9.4";
+
+  src = fetchurl {
+    url = "https://github.com/CatalystDevOrg/Catalyst/releases/download/v${version}/catalyst-${version}.AppImage";
+    hash = "sha256-6t1RAxmRc/1fAQT4Qnd42kh3cxgRZr74k8gwebTb0Ic=";
+    name = "${pname}-${version}.AppImage";
+  };
+
+  buildInputs = [
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/catalyst/package.nix:25]:
```suggestion
  nativeBuildInputs = [
```

File: pkgs/by-name/ca/catalyst/package.nix [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  appimageContents = appimageTools.extractType2 {
+    name = "${pname}-${version}";
+    inherit src;
+  };
+
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [ makeWrapper ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/catalyst/package.nix:38]:
```suggestion
  nativeBuildInputs = [ makeWrapper asar ];
```
or on two lines, depending on nixfmt

File: pkgs/by-name/ca/catalyst/package.nix [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/CatalystDevOrg/Catalyst/releases/download/v${version}/catalyst-${version}.AppImage";
+    hash = "sha256-6t1RAxmRc/1fAQT4Qnd42kh3cxgRZr74k8gwebTb0Ic=";
+    name = "${pname}-${version}.AppImage";
+  };
+
+  buildInputs = [
+    asar
+  ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/catalyst/package.nix:27]:
```suggestion
```

**Comment by jdev082** [comment]
All requested changes have been made.

**Comment by jdev082** [comment]
Not ready to be merged as I'm deciding whether or not to rename this package to `catalyst-browser`. I have a two reasons:

- So it's grouped with the other packages on repology
- There are many other projects and applications called Catalyst

The decision and changes will be made tomorrow.

**Comment by jdev082** [comment]
Package rename has been completed.

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/by-name/ca/catalyst-browser/package.nix [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+  src = fetchurl {
+    url = "https://github.com/CatalystDevOrg/Catalyst/releases/download/v${version}/catalyst-${version}.AppImage";
+    hash = "sha256-6t1RAxmRc/1fAQT4Qnd42kh3cxgRZr74k8gwebTb0Ic=";
+    name = "catalyst-${version}.AppImage";
+  };
+
+  appimageContents = appimageTools.extractType2 {
+    name = "catalyst-${version}";
+    inherit src;
+  };
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/catalyst-browser/package.nix:28]:
```suggestion
  appimageContents = appimageTools.extractType2 {
    inherit pname src version;
  };
```

**Comment by jdev082** [comment]
Requested changes have been applied.

**Comment by jdev082** [comment]
It seems like I've broken the PR so I'll have to create a new one.

**Comment by jdev082** [comment]
Unless this fixes it

File: pkgs/by-name/ca/catalyst-browser/package.nix [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+  electron = electronPackage;
+in
+stdenv.mkDerivation rec {
+  pname = "catalyst-browser";
+  version = "3.9.4";
+
+  src = fetchurl {
+    url = "https://github.com/CatalystDevOrg/Catalyst/releases/download/v${version}/catalyst-${version}.AppImage";
+    hash = "sha256-6t1RAxmRc/1fAQT4Qnd42kh3cxgRZr74k8gwebTb0Ic=";
+    name = "catalyst-${version}.AppImage";
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/catalyst-browser/package.nix:22]: This looks like a no-op? (the name parameter is optional)

**Pull request merged**